packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -543,16 +562,25 @@ - (void)setPlaybackSpeed:(double)speed {
 }
 
 - (CVPixelBufferRef)copyPixelBuffer {
+  // Ensure video sampling at regular intervals. This function is not called at exact time intervals
+  // so CACurrentMediaTime returns irregular timestamps which causes missed video frames. The range
+  // outside of which targetTime is reset should be narrow enough to make possible lag as small as
+  // possible and at the same time wide enough to avoid too frequent resets which would lead to
+  // irregular sampling. Ideally there would be a targetTimestamp of display link used by flutter
+  // engine (FlutterTexture can provide timestamp and duration or timestamp and targetTimestamp).
+  CFTimeInterval currentTime = CACurrentMediaTime();
+  if (fabs(self.targetTime - currentTime) > self.frameUpdater.duration / 2) {
```

Comment:
There is an explanation in that comment that it should be narrow enough and at the same time wide enough `/2` is thus a compromise that worked well (for example `/3` worked worse and `/2` worked as good as anything higher). Yes it is for example to account for possible drifting or to reset after pause. Just adding some value to some time at display refresh intervals does not guarantee that it will be in sync actual time (and not just because of finite precision of the type).

Reply:
> There is an explanation in that comment that it should be narrow enough and at the same time wide enough

Yes, I read that, and still didn't know why `/ 2` was the resulting code. If it was determined empirically, that should be clearly stated in the comment, with a brief explanation of how.